modules/api/src/main/java/org/apache/fluo/api/config/FluoConfiguration.java

```diff
@@ -525,31 +714,41 @@ public void print() {
     }
   }
 
-  private boolean verifyStringPropSet(String key) {
-    if (containsKey(key) && !getString(key).isEmpty()) {
-      return true;
+  private boolean verifyStringPropSet(String... keys) {
+    for (String key : keys) {
+      if (containsKey(key) && !getString(key).isEmpty()) {
+        return true;
+      }
     }
-    log.info(key + " is not set");
+    log.info(keys[0] + " is not set");
     return false;
   }
 
-  private boolean verifyStringPropNotSet(String key) {
-    if (containsKey(key) && !getString(key).isEmpty()) {
-      log.info(key + " should not be set");
-      return false;
+  private boolean verifyStringPropNotSet(String... keys) {
+    for (String key : keys) {
+      if (containsKey(key) && !getString(key).isEmpty()) {
+        log.info(key + " should not be set");
+        return false;
+      }
     }
     return true;
   }
 
+  public boolean hasRequiredConnectionProps() {
+    boolean valid = true;
+    valid &= verifyStringPropSet(CONNECTION_APPLICATION_NAME_PROP, CLIENT_APPLICATION_NAME_PROP);
+    return valid;
+  }
+
   /**
    * Returns true if required properties for FluoClient are set
    */
   public boolean hasRequiredClientProps() {
     boolean valid = true;
-    valid &= verifyStringPropSet(CLIENT_APPLICATION_NAME_PROP);
-    valid &= verifyStringPropSet(CLIENT_ACCUMULO_USER_PROP);
-    valid &= verifyStringPropSet(CLIENT_ACCUMULO_PASSWORD_PROP);
-    valid &= verifyStringPropSet(CLIENT_ACCUMULO_INSTANCE_PROP);
+    valid &= verifyStringPropSet(CONNECTION_APPLICATION_NAME_PROP, CLIENT_APPLICATION_NAME_PROP);
+    valid &= verifyStringPropSet(ACCUMULO_USER_PROP, CLIENT_ACCUMULO_USER_PROP);
```

Comment:
Take a look at the FluoClientImpl constructor.  I was no longer using `config.hasRequiredClientClientProps()` to check user input but rather sanity check the configuration after mergeZookeeper method is called.  Let me know what you think..